king_phisher/client/windows/plugin_manager.py

```diff
@@ -178,18 +178,23 @@ def _load_catalogs(self, refresh=False):
 					continue
 			self.logger.debug("downloading catalog: {}".format(catalog_url))
 			self._update_status_bar("Loading, downloading catalog: {}".format(catalog_url))
-			try:
-				catalog = Catalog.from_url(catalog_url)
-			except requests.exceptions.ConnectionError:
-				self.logger.warning("connection error trying to download catalog url: {}".format(catalog))
-				self.idle_show_dialog_error('Catalog Loading Error', "Failed to download catalog, check your internet connection.")
-			except Exception:
-				self.logger.warning("failed to add catalog by url", exc_info=True)
-				self.idle_show_dialog_error('Catalog Loading Error', "Failed to add catalog")
-			else:
-				self.catalog_plugins.add_catalog(catalog, catalog_url=catalog_url, cache=True)
+			catalog = self._load_catalog_from_url(catalog_url)
+			if not catalog:
+				continue
+			self.catalog_plugins.add_catalog(catalog, catalog_url=catalog_url, cache=True)
 		self._load_plugins()
 
+	def _load_catalog_from_url(self, catalog_url):
+		try:
+			catalog = Catalog.from_url(catalog_url)
+		except requests.exceptions.ConnectionError:
+			self.logger.warning("connection error trying to download catalog url: {}".format(catalog_url))
+			self.idle_show_dialog_error('Catalog Loading Error', "Failed to download catalog, check your internet connection.")
+		except Exception:
+			self.logger.warning("failed to add catalog by url", exc_info=True)
+			self.idle_show_dialog_error('Catalog Loading Error', "Failed to add catalog")
+		return catalog
```

Comment:
This should be in an else block or the except blocks should end in `return None` otherwise `return catalog` will error out because `catalog` is undefined.